allennlp/nn/decoding/maximum_marginal_likelihood.py

```diff
@@ -33,43 +34,50 @@ def decode(self,
                decode_step: DecoderStep,
                targets: torch.Tensor = None,
```

Comment:
These arguments aren't optional at the moment, either raise a `NotImplementedError` or remove.

Reply:
Done.